frontend/src/components/SignUp.jsx

```diff
@@ -1,43 +1,44 @@
-import { useState } from "react";
+import { Component, useState } from "react";
 import "./SignUp.module.css";
 import logo from "../assets/icons/hooli-logo.png";
 
-function SignUp() {
-  let token = "";
-  const [data, setData] = useState({
-    name: "",
-    email: "",
-    password: "",
-  });
+class SignUp extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { name: "", email: "", password: "" };
 
-  const handleChange = ({ currentTarget: input }) => {
-    setData({ ...data, [input.name]: input.value });
-  };
-  const options = {
-    method: "POST",
-    headers: {
-      "Content-Type": "application/json",
-    },
-    body: JSON.stringify({
-      name: data.name,
-      email: data.email,
-      password: data.password,
-    }),
-  };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
 
-  const handleSubmit = (e) => {
+  handleChange(event) {
+    event.preventDefault();
+    const target = event.target;
+    this.setState({
+      [target.name]: target.value,
+    });
+  }
+
+  handleSubmit = (e) => {
     e.preventDefault();
     console.log(
       JSON.stringify({
-        name: data.name,
-        email: data.email,
-        password: data.password,
+        name: this.state.name,
+        email: this.state.email,
+        password: this.state.password,
       })
     );
-    console.log(data);
-    console.log(options.body);
-    //console.log(process.env.PORT);
-    fetch("/api/users", options)
+    fetch("/api/users", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        name: this.state.name,
+        email: this.state.email,
+        password: this.state.password,
+      }),
+    })
       .then((response) => {
         if (response.status === 201) {
           console.log("Sign up successful!");
```

Comment:
At this point, could you also store the token with `token = response.body.user.tokens[0].token`